File: src/main/java/org/shingas/viewerSMP/utils/ConfigManager.java
package org.shingas.viewerSMP.utils;


import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.plugin.java.JavaPlugin;
import org.shingas.viewerSMP.ViewerSMP;
import org.shingas.viewerSMP.data.CustomConfig;

public class ConfigManager {

    private final JavaPlugin plugin;

    private CustomConfig mainConfig;
    private CustomConfig cosmetics;
    private CustomConfig feelings;

    public ConfigManager(ViewerSMP plugin) {
        this.plugin = plugin;
    }

    public void loadConfigs() {
        plugin.saveDefaultConfig(); // config.yml

        mainConfig = new CustomConfig(plugin, "config.yml");
        cosmetics = new CustomConfig(plugin, "cosmetics.yml");
        feelings = new CustomConfig(plugin, "feelings.yml");
    }

    public void reloadAll() {
        plugin.reloadConfig();
        mainConfig.reload();
        cosmetics.reload();
        feelings.reload();
    }

    public FileConfiguration config() {
        return mainConfig.get();
    }

    public FileConfiguration cosmetics() {
        return cosmetics.get();
    }

    public FileConfiguration feelings() {
        return feelings.get();
    }
}

File: src/main/kotlin/com/shingaspt/viewerSMP/utils/ConfigManager.kt
package com.shingaspt.viewerSMP.utils

import com.shingaspt.viewerSMP.ViewerSMP
import org.bukkit.Bukkit
import org.bukkit.configuration.file.FileConfiguration

class ConfigManager {

    private lateinit var config: FileConfiguration
    private var plugin = ViewerSMP.instance

    init {
        setupConfig()
    }

    fun setupConfig() {
        // Only saves default config if it doesn't exist yet
        plugin.saveDefaultConfig()
        config = plugin.config
    }

    fun reload() {
        plugin.reloadConfig()
        config = plugin.config
        Bukkit.getLogger().info("Ticket's config was reloaded.")
    }

    fun getString(path: String): String {
        return config.getString(path).toString()
    }

    fun getInt(path: String): Int {
        return config.getInt(path)
    }

    fun pathExists(path: String): Boolean {
        return config.contains(path)
    }

    fun save() {
        plugin.saveConfig()
        Bukkit.getLogger().info("Ticket's config was saved.")
    }

}